File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/service/ApiKeyService.java
package at.ac.tuwien.sepr.groupphase.backend.service;

import at.ac.tuwien.sepr.groupphase.backend.endpoint.dto.ApiKeyDtoCreate;
import at.ac.tuwien.sepr.groupphase.backend.endpoint.dto.ApiKeyDtoFirst;
import at.ac.tuwien.sepr.groupphase.backend.endpoint.dto.ApiKeyPageDto;
import org.springframework.data.domain.Pageable;

public interface ApiKeyService {
    /**
     * Get a list of API keys.
     *
     * @param pageable the page information
     * @return a list of API keys
     */
    ApiKeyPageDto getApiKeys(Pageable pageable);

    /**
     * Create a new API key.
     *
     * @param apiKeyDtoCreate the API key to create
     * @return the created API key
     */
    ApiKeyDtoFirst createApiKey(ApiKeyDtoCreate apiKeyDtoCreate);

    /**
     * Delete an API key.
     *
     * @param id the API key id to delete
     */
    void deleteApiKey(long id);
}

File: frontend/src/app/services/api-key.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Globals} from "../global/globals";
import {ApiKeyDto, ApiKeyDtoFirst, ApiKeyPageDto} from "../dtos/api-keys";

@Injectable({
  providedIn: 'root'
})
export class ApiKeyService {

  private apiKeyUri: string = this.globals.backendUri + '/api-keys';

  constructor(
    private http: HttpClient,
    private globals: Globals
  ) { }

  /**
   * Creates a new api key
   *
   * @param apiKeyCreateDto
   */
  public createApiKey(apiKeyCreateDto: any) {
    return this.http.post<ApiKeyDtoFirst>(this.apiKeyUri, apiKeyCreateDto);
  }

  /**
   * Get all api keys from the backend.
   *
   * @return an Observable of the api keys
   */
  getApiKeysAll(page: number, size: number) {
    return this.http.get<ApiKeyPageDto>(`${this.apiKeyUri}?page=${page}&size=${size}`);
  }

  /**
   * Deletes an api key by id
   *
   * @param id
   */
  deleteApiKey(id: number) {
    return this.http.delete(`${this.apiKeyUri}/${id}`);
  }
}
